### PR TITLE
fix: fix card issues

### PR DESCRIPTION
**Pull request by burakukula**
Small fixes for keyboard navigation for Card.
Removing `isFocused` property - are there any cases where we actually need to control this property?

### CODE REVIEW

File: packages/components/card/src/base-card/BaseCard.styles.ts [hunk 1]
```diff
@@ -40,6 +40,12 @@ export const getBaseCardStyles = () => {
         textDecoration: 'none',
         transition: `border-color ${tokens.transitionDurationDefault} ${tokens.transitionEasingDefault},
     box-shadow ${tokens.transitionDurationShort} ${tokens.transitionEasingDefault}`,
+
+        '&:focus': css({
```
Comment (massao) [inline packages/components/card/src/base-card/BaseCard.styles.ts:44]: shouldn't we do the same for `&:hover`?

File: packages/components/card/src/base-card/BaseCard.tsx [hunk 2]
```diff
@@ -66,7 +66,6 @@ function _BaseCard<E extends React.ElementType = typeof BASE_CARD_DEFAULT_TAG>(
   forwardedRef: React.Ref<HTMLElement>,
 ) {
   const styles = getBaseCardStyles();
-  const [isFocused, setIsFocused] = useState(isFocusedProp ?? false);
```
Comment (massao) [inline packages/components/card/src/base-card/BaseCard.tsx:69]: We can also remove the `isFocusProp` from the props list, if it's not used anywhere else.

File: packages/components/card/src/base-card/BaseCard.styles.ts (hunk 1 above)
Comment (burakukula) [reply packages/components/card/src/base-card/BaseCard.styles.ts:44]: hover is actually depending on the fact if the card is interactive or not, so if it's a button. Maybe it's easier to keep it as it is in that case?

**Comment by denkristoffer** [comment]
My note here would be that without this (and other states) we lose the ability to check them with Chromatic.

**Comment by burakukula** [comment]
> My note here would be that without this (and other states) we lose the ability to check them with Chromatic.

ahhh, ok, I understand, but should we keep it for this purpose?

**Comment by denkristoffer** [comment]
Don't know 🤷 I found it helpful when building the components but there's no denying that it complicates something simple.

**Comment by massao** [comment]
> Don't know 🤷 I found it helpful when building the components but there's no denying that it complicates something simple.

We can also check if there isn't any addon for storybook that would help with that, the one I know of is this one: https://storybook.js.org/addons/storybook-addon-pseudo-states
but it would required a separate story for different states, as far as I remember.

**Comment by burakukula** [comment]
> > Don't know 🤷 I found it helpful when building the components but there's no denying that it complicates something simple.
> 
> We can also check if there isn't any addon for storybook that would help with that, the one I know of is this one: https://storybook.js.org/addons/storybook-addon-pseudo-states but it would required a separate story for different states, as far as I remember.

@massao @denkristoffer maybe let's figure this out separately since that's something that might also apply to other components like Buttons?
If we agree I will clean up stories and remove the focus state from the overview.

**Review by massao — APPROVED** [review]

**Pull request merged**